packages/@tailwindcss-upgrade/src/codemods/migrate-at-config.ts

```diff
@@ -0,0 +1,73 @@
+import path from 'node:path'
+import { AtRule, type Plugin, type Root } from 'postcss'
+import type { Stylesheet } from '../stylesheet'
+import { walk, WalkAction } from '../utils/walk'
+
+export function migrateAtConfig(
+  sheet: Stylesheet,
+  { configFilePath }: { configFilePath: string },
+): Plugin {
+  function migrate(root: Root) {
+    let hasConfig = false
+    root.walkAtRules('config', () => {
+      hasConfig = true
+      return false
+    })
+
+    // We already have a `@config`
+    if (hasConfig) return
+
+    // We don't have a sheet with a file path
+    // TODO: Why? Tests?
+    if (!sheet.file) return
+
+    // Should this sheet have an `@config`?
+    // 1. It should be a root CSS file
+    if (sheet.parents.size > 0) return
+
+    // 2. It should include a `@import "tailwindcss"`
+    let hasTailwindImport = false
+    root.walkAtRules('import', (node) => {
+      if (node.params.includes('tailwindcss')) {
+        hasTailwindImport = true
+        return false
+      }
+    })
+    if (!hasTailwindImport) return
+
+    // Figure out the path to the config file
+    let sheetPath = sheet.file
+    let configPath = configFilePath
+
+    let relativePath = path.relative(path.dirname(sheetPath), configPath)
+    if (relativePath[0] !== '.') {
+      relativePath = `./${relativePath}`
+    }
```

Comment:
We'll want to test this on Windows. I think this might be kinda broken there?

Reply:
I think we decided that all these globs should be unix paths with `/`. But _maybe_ the `relativePath` itself might be wrong then. Running the tests as-is on Windows right now.

Reply:
yeah I meant `path.relative` is wrong. It uses Windows separators on windows.